### PR TITLE
Recipe app

### PR DESCRIPTION
**Pull request by tugsem**
## Project requirements:

- You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
Login page and registration page:
- Should be built with Devise.
Food list:
- Should display a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also quantity of a given food).
- Should lead to a form that allows users to add new food.
Recipes list:
- Should display a list of recipes created by the logged-in user as in the wireframe.
- Should lead to recipe details.
- If the user is the owner of the recipe, should allow the user to delete it.
Public recipe list:
- Should display a list of all public recipes ordered by newest as in the wireframe.
- Should lead to recipe details.
Recipe details:
- If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
- If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
General shopping list view (only for a group that has less than 3 members):
- Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
- Should count the total food items and total price of the missing food.
Make sure there are no N+1 queries happening.
Create a navigation menu that allows users to open all of the pages you created.

### CODE REVIEW

File: app/controllers/recipes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+class RecipesController < ApplicationController
+  def index
+    @user = User.find(current_user.id)
+    @recipes = Recipe.where(user_id: @user.id)
+  end
+
+  def show
+    @recipe = Recipe.find(params[:id])
+    @recipe_foods = @recipe.recipe_foods
+  end
```
Comment (gemmen29) [inline app/controllers/recipes_controller.rb:10]: - [ ] Kindly fix the `n+1` query that happens in that action `Recipes#show` as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md#:~:text=Make%20sure%20there%20are%20no%20N%2B1%20queries%20happening.).

File: app/controllers/public_recipes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class PublicRecipesController < ApplicationController
+  def index
+    @public_recipes = Recipe.where(public: true).order(created_at: :desc)
+  end
```
Comment (gemmen29) [inline app/controllers/public_recipes_controller.rb:4]: - [ ] Kindly fix the `n+1` query that happens in that action `PublicRecipes#index` as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md#:~:text=Make%20sure%20there%20are%20no%20N%2B1%20queries%20happening.).

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello  @TEAM          , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- [x] Your project is professional 

- [x]  Your repo is professional 




 

- [x] No Linter Errors





Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**